evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -115,7 +115,7 @@ def add_result(self, score_to_minimize, pipeline, trained_pipeline_results):
             self._first_batch_results.append((score_to_minimize, pipeline.__class__))
 
         current_best_score = self._best_pipeline_info.get(pipeline.model_family, {}).get('score', np.inf)
-        if score_to_minimize is not None and score_to_minimize < current_best_score:
+        if score_to_minimize is not None and score_to_minimize < current_best_score and pipeline.model_family != ModelFamily.ENSEMBLE:
```

Comment:
We don't want to add the ensemble model, which uses `self._best_pipeline_info` for input pipelines, to _best_pipeline_info.

Reply:
I believe it would have "worked" but the second ensemble would try to add an ensemble as an input pipeline, which is probably not what we want :d